src/operation-node/case-node.ts

```diff
@@ -0,0 +1,59 @@
+import { freeze } from '../util/object-utils.js'
+import { OperationNode } from './operation-node.js'
+import { WhenNode } from './when-node.js'
+
+export interface CaseNode extends OperationNode {
+  readonly kind: 'CaseNode'
+  readonly value?: OperationNode
+  readonly when?: ReadonlyArray<WhenNode>
+  readonly else?: OperationNode
+  readonly isStatement?: boolean
+}
+
+/**
+ * @internal
+ */
+export const CaseNode = freeze({
+  is(node: OperationNode): node is CaseNode {
+    return node.kind === 'CaseNode'
+  },
+
+  create(value?: OperationNode): CaseNode {
+    return freeze({
+      kind: 'CaseNode',
+      value,
+    })
+  },
+
+  cloneWithWhen(caseNode: CaseNode, when: WhenNode): CaseNode {
+    return freeze({
+      ...caseNode,
+      when: caseNode.when ? [...caseNode.when, when] : [when],
```

Comment:
`freeze` the lists, since that's what's done in other nodes. I'm been thinking that maybe we could live without freezing stuff. But until we drop it, let's be consistent.

Reply:
Good catch!

We could drop these and add a strong disclaimer in the docs - if you hack it, we don't bail you out when you f things up.